src/include/common/file_system/file_system.h

```diff
@@ -43,6 +43,12 @@ class KUZU_API FileSystem {
 
     virtual bool canHandleFile(const std::string& /*path*/) const { KU_UNREACHABLE; }
 
+    std::string pathSeparator();
+
+    std::string extractName(const std::string& path);
```

Comment:
comment what this function is doing. Extract name for a given path is ambiguous.